### ISSUE TITLE
120% cpu when the only destination isnt available

### ISSUE THREAD

**Issue by bangert** [elastic/libbeat#270]
Hi,

when just configuring a single destination for a filebeat, filebeat goes into 120% cpu if said destination isnt available. on the network i can see lots of syn packets flying past.

the following config illustrates the point:
```yaml
filebeat:
  prospectors:
       - 
         paths:
            - "/var/log/messages"
output:
  logstash:
    enabled: true
    hosts:
       - "127.0.0.1:23456"
    loadbalance: false
    tls:
       disabled: true
```
Make sure logstash isnt running on 127.0.0.1:23456.

In case of a hostname instead of an ip address, filebeat will ony consume 65% cpu presumably because the DNS lookup takes a long time.

This has an easy workaround: simply duplicating the host will make filebeat behave more sane.

Basically, it looks like there is some ratelimiting which is bypassed in the case of a single destination.

This is filebeat version 1.0.0-beta4 (amd64).
Thanks

**Comment by andrewkroh** [comment]
Hi @bangert, check out [this bug](https://github.com/elastic/filebeat/issues/144#issuecomment-153191910) in the filebeat repository which is now fixed and will be released as rc1 soon.

**Comment by bangert** [comment]
thanks. sorry for the dupe
